activation/poet.go

```diff
@@ -257,13 +262,15 @@ func (c *HTTPPoetClient) req(ctx context.Context, method, path string, reqBody,
 	if res.StatusCode != http.StatusOK {
 		c.logger.Info("got poet response != 200 OK", zap.String("status", res.Status), zap.String("body", string(data)))
 	}
-	switch res.StatusCode {
-	case http.StatusOK:
-	case http.StatusNotFound:
+	switch {
+	case res.StatusCode == http.StatusOK:
+	case res.StatusCode == http.StatusNotFound:
 		return fmt.Errorf("%w: response status code: %s, body: %s", ErrNotFound, res.Status, string(data))
-	case http.StatusServiceUnavailable:
+	case res.StatusCode == http.StatusServiceUnavailable:
 		return fmt.Errorf("%w: response status code: %s, body: %s", ErrUnavailable, res.Status, string(data))
-	case http.StatusBadRequest:
+	case res.StatusCode == http.StatusBadRequest && strings.Contains(string(data), registration.ErrTooLateToRegister.Error()):
+		return fmt.Errorf("%w: response status code: %s, body: %s", ErrSubmitTooLate, res.Status, string(data))
+	case res.StatusCode == http.StatusBadRequest:
```

Comment:
Instead of splitting these two into their own cases, I think I would leave `switch res.StatusCode` and just have a `if` in the `case http.StatusBadRequest`.